cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/aggregation/DeduplicateAggregationDefinition.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.aggregation;
+
+import io.cdap.cdap.etl.api.relational.Expression;
+
+import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Specifies how a deduplicate operation should be executed.
+ * A deduplication is an operation that takes in a set of rows and for each group of rows for which the values for
+ * certain group by fields are the same, outputs a single row.
+ * This single row can be determined by performing an operation on the set of rows which returns a single row.
+ * Currently the two supported operations are returning the row with the MIN or MAX value of a particular expression.
+ */
+public class DeduplicateAggregationDefinition extends AggregationDefinition {
+  List<SimpleEntry<Expression, FilterFunction>> filterExpressions;
+
+  private DeduplicateAggregationDefinition(List<Expression> groupByExpressions,
+                                           List<Expression> selectExpressions,
+                                           List<SimpleEntry<Expression, FilterFunction>> filterExpressions) {
+    super(groupByExpressions, selectExpressions);
+    this.filterExpressions = filterExpressions;
+  }
+
+  /**
+   * Get the list of expressions using which a single record will be selected from a group of records for deduplication.
+   * Each entry in the list is defined as a pair of an {@link Expression} and {@link FilterFunction}, specifying
+   * the function to be executed on the expression.
+   * @return A {@link List} of {@link SimpleEntry} objects, each of which is a mapping between an {@link Expression}
+   * and a {@link FilterFunction}.
```

Comment:
I'd not say it's a mapping. It's not used as a map, rather as two parameters to select one row out of the group

Reply:
Fixed.